.orchestra/ci/binary-archives-hook.sh

```diff
@@ -79,15 +79,16 @@ if [ "${#REDIST_PATHS[@]}" -gt 0 ]; then
     # Build docker image
     #
 
+    # TODO: check again when upgrading to 24.04 if the 'sudo' can be dropped
     LOCAL_IMAGE="localhost/revng-image-$(tr -dc A-Za-z0-9 < /dev/urandom | head -c 16)"
-    podman build \
+    sudo -i podman build \
         -t "$LOCAL_IMAGE" \
         -f "$ORCHESTRA_DOTDIR/support/Dockerfile.binary-archives" \
         "$DOCKER_CONTEXT"
-    podman login -u "$PODMAN_REGISTRY_USER" \
-                 -p "$PODMAN_REGISTRY_PASSWORD" \
-                 "$(cut -d/ -f1 <<< "$PODMAN_IMAGE_TARGET")"
-    podman push "$LOCAL_IMAGE" "$PODMAN_IMAGE_TARGET"
+    sudo -i podman login -u "$PODMAN_REGISTRY_USER" \
+        -p "$PODMAN_REGISTRY_PASSWORD" \
```

Comment:
This might leak in logs.

Reply:
The alternative is to have an `auth.json`, but even in that case there would be a
```
echo "$USER:$PASSWORD" > auth.json
```
which would have the same problem
Another option is piping from stdin, but that as well has the same issue

Reply:
Does gitlab protect us from leaking?

Reply:
Not at the moment, all variables are pushed by the ci-hook without `masked: true`, however masking has some limitations (no multiline, max length limit) which won't help with e.g. `S3CMD_CONFIG` which is multiline and with credentials.

Reply:
Dunno, I'd work under the assumption that eventually we *will* run this code with `set -x`.
We could:
 ```
USER=a PASS=b python -c 'from os import getenv as env; print(f"""{env("USER")}:{env("PASS")}""")'
```

Reply:
Ok, but that seems too specific to this line of code, we still have credentials elsewhere
(That's beside the fact that the line in question is quite ugly)

Reply:
> Ok, but that seems too specific to this line of code, we still have credentials elsewhere

Eh, let's backlog fixing this.

>  (That's beside the fact that the line in question is quite ugly)

Eh, we can make a `bash` function `print-env '{env("USER")}:{env("PASS")}'`.

Reply:
> Eh, let's backlog fixing this.

Ok

> Eh, we can make a bash function print-env '{env("USER")}:{env("PASS")}'.

I was thinking of a similar approach, agreed